cpu/samd21/periph/rtt.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_samd21
+ * @{
+ * @file
+ * @brief       Low-level RTT driver implementation
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @}
+ */
+
+#include <time.h>
+#include "cpu.h"
+#include "periph/rtt.h"
+#include "periph_conf.h"
+#include "sched.h"
+#include "thread.h"
+
+/* guard file in case no RTT device was specified */
+#if RTT_NUMOF
+
+
+typedef struct {
+    rtt_cb_t    overflow_cb;    /**< called from RTT interrupt on overflow */
+    void*       overflow_arg;   /**< argument passed to overflow callback */
+
+    rtt_cb_t    alarm_cb;       /**< called from RTT interrupt on alarm */
+    void*       alarm_arg;      /**< argument passen to alarm callback */
+} rtt_state_t;
+
+static rtt_state_t rtt_callback;
+
+
+/**
+ * @brief Initialize RTT module
+ */
+void rtt_init(void)
+{
+    RtcMode0 *rtcMode0 = &(RTT_DEV);
+
+    /* Turn on power manager for RTC */
+    PM->APBAMASK.reg |= PM_APBAMASK_RTC;
+
+    /* RTC uses External 32,768KHz Oscillator because OSC32K isn't accurate
+     * enough (p1075/1138). Also keep running in standby. */
+    SYSCTRL->XOSC32K.reg =  SYSCTRL_XOSC32K_ONDEMAND |
+                            SYSCTRL_XOSC32K_RUNSTDBY |
+                            SYSCTRL_XOSC32K_EN32K |
+                            SYSCTRL_XOSC32K_XTALEN |
+                            SYSCTRL_XOSC32K_STARTUP(6) |
+                            SYSCTRL_XOSC32K_ENABLE;
+
+    /* Setup clock GCLK2 with divider 1 */
+    GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) | GCLK_GENDIV_DIV(1);
+    while (GCLK->STATUS.bit.SYNCBUSY);
+
+    /* Enable GCLK2 with XOSC32K as source. Use divider without modification
+     * and keep running in standby. */
+    GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(2) |
+                        GCLK_GENCTRL_GENEN |
+                        GCLK_GENCTRL_SRC_XOSC32K |
+                        GCLK_GENCTRL_RUNSTDBY;
+    while (GCLK->STATUS.bit.SYNCBUSY);
+
+    /* Connect GCLK2 to RTC */
+    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_GEN_GCLK2 |
+                        GCLK_CLKCTRL_CLKEN |
+                        GCLK_CLKCTRL_ID(RTC_GCLK_ID);
+    while (GCLK->STATUS.bit.SYNCBUSY);
+
+    /* Disable RTC */
+    rtt_poweroff();
+
+    /* Reset RTC */
+    rtcMode0->CTRL.bit.SWRST = 1;
+    while (rtcMode0->STATUS.bit.SYNCBUSY || rtcMode0->CTRL.bit.SWRST);
+
+    /* Configure as 32bit counter with no prescaler and no clear on match compare */
+    rtcMode0->CTRL.reg = RTC_MODE0_CTRL_MODE_COUNT32 |  RTC_MODE0_CTRL_PRESCALER_DIV1;
```

Comment:
As we said, because we can't use both RTT and RTC, a ``PRESCALER`` could be set here. periph_conf.h is a good place to set the configuration. 

Same with GENDIV, use ``GCLK_GENDIV_DIV(4)``